### PR TITLE
Upgrade to webpack 5.73.0

### PR DESCRIPTION
**Pull request by LunaticMuch**
This PR includes the update to webpack 5.73.0

Because of a new bundle optimisation, the old approach to `eval` the bundle file did not work anymore. [This is intended](https://github.com/webpack/webpack/issues/15928) with webpack 5.22+.
I tweak, thanks to a suggestion, the bundle evaluation function using `require`. Node behaviour is to cache imported module and invalidating/clearing the cache is not always trivial. 
Hence, I made a change and now each bundle is generated using the `[fullhash]` which makes the file name always different. 

All tests are passed.

After merging, the npm package must be also rebuilt to ensure the new library is distributed. I moved the version to `1.2.3`

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        "--inspect-brk",
+        "${workspaceRoot}/node_modules/.bin/jest",
+        "--runInBand"
+      ],
+      "console": "integratedTerminal",
+      "internalConsoleOptions": "neverOpen",
+      "port": 9229
+    }
+  ]
+}
```
Comment (bhovhannes) [inline .vscode/launch.json:18]: please add `.vscode` folder to `.gitignore`.  
Not everyone uses VSCode, including myself and its config is a personal thing which does not always make sense to everyone.

File: package.json [hunk 2]
```diff
@@ -1,15 +1,15 @@
 {
   "name": "svg-as-symbol-loader",
-  "version": "1.2.2",
+  "version": "1.2.3",
```
Comment (bhovhannes) [inline package.json:3]: Do not manually change version here. It will be bumped during publishing automatically.

File: package.json [hunk 3]
```diff
@@ -30,12 +30,13 @@
   },
   "homepage": "https://github.com/bhovhannes/svg-as-symbol-loader#readme",
   "devDependencies": {
-    "@arkweid/lefthook": "0.7.7",
     "@bhovhannes/shared-config": "0.0.1",
+    "@evilmartians/lefthook": "^0.8.0",
     "@jest/globals": "28.1.0",
-    "jest": "28.1.0",
+    "clear-module": "^4.1.2",
```
Comment (bhovhannes) [inline package.json:36]: Why do we need this dependency? I don't see it is used somewhere.

File: test/loader.spec.js [hunk 4]
```diff
@@ -1,219 +1,196 @@
...
+  function getBundleFile(assetName) {
+    return path.join(outputDir, assetName)
+  }
+  // require is now mandatory with webpack >5.22.0
+  return eval(`require("${getBundleFile(assetName)}")`)
 }
 
-describe("svg-as-symbol-loader", function () {
-  // Clean generated cache files before each test so that we can call each test with an empty state.
-  afterEach(() => fsPromises.unlink(getBundleFile()));
```
Comment (bhovhannes) [inline test/loader.spec.js:59]: Does removing this mean that after each test run some files will be generated and left on a filesystem?

If so, can you please change webpack config to store them inside some folder and remove that folder in `afterAll` ?

File: test/loader.spec.js [hunk 5]
```diff
@@ -1,219 +1,196 @@
...
-
-function getBundleFile() {
-  return path.join(outputDir, bundleFileName);
-}
+const { describe, it, afterEach, expect, afterAll } = require('@jest/globals')
+const { rejects } = require('assert')
+// const util = require('util');
+// const fs = require('fs');
+// const readdir = util.promisify(fs.readdir);
+// const unlink = util.promisify(fs.unlink);
```
Comment (bhovhannes) [inline test/loader.spec.js:6]: Please remove these comments

File: src/loader.js [hunk 6]
```diff
@@ -1,102 +1,97 @@
-const loaderUtils = require("loader-utils");
+const loaderUtils = require('loader-utils')
```
Comment (bhovhannes) [inline src/loader.js:1]: One general question. There are many formatting changes. Did you run `npm run format` before commit?

**Review by bhovhannes — COMMENTED** [review]
Thank you for figuring this out!
I've left a few comments, please have a look.

File: package.json (hunk 3 above)
Comment (LunaticMuch) [reply package.json:36]: We do not, I removed it, thought I did it but apparently I did not.

File: test/loader.spec.js (hunk 4 above)
Comment (LunaticMuch) [reply test/loader.spec.js:59]: The bundle file generated is different now for any test. This solves the trouble with module cache. 
The new `clean: true` will clean the output directory, and the asset name will have the hash in the name so it won't be possible to recycle the same bundle.

File: src/loader.js (hunk 6 above)
Comment (bhovhannes) [reply src/loader.js:1]: I think I've never run prettier after renovate bumped its version. And you run it :) That's why so much changes. But that's fine, I was just curious if it was prettier or IDE formatting.

Comment (LunaticMuch) [reply src/loader.js:1]: 😄 I tried also running format with my VSCode and returned the same output. Single quote is the new way to go 👍🏻

**Comment by LunaticMuch** [comment]
> Thank you for figuring this out!
> 
> I've left a few comments, please have a look.

Think it's all sorted now. I can squash everything if you wish.

**Comment by bhovhannes** [comment]
Released new version in [1.2.3](https://github.com/bhovhannes/svg-as-symbol-loader/releases/tag/v1.2.3)

Thank you for figuring this out! It was a long standing issue here.

**Pull request merged**